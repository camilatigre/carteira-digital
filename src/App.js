import React from "react";
import Login from "./pages/Login";
import Register from "./pages/Register";
import Home from "./pages/Home";
import Unauthorized from "./pages/Unauthorized";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import { GlobalProvider } from "./context/GlobalState";
import "./styles/Main.css";

const App = () => {
  return (
    <GlobalProvider>
      <Router>
        <Switch>
          <Route path="/" component={Login} exact />
          <Route path="/register" component={Register} exact />
          <Route path="/home" component={Home} exact />
          <Route path="/401" component={Unauthorized} exact />
        </Switch>
      </Router>
    </GlobalProvider>
  );
};

export default App;
