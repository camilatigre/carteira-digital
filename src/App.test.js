import React from 'react';
import App from './App';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

it('Deveria renderizar o App', () => {
  const app = shallow(<App />);
  expect(toJson(app)).toMatchSnapshot();
});
