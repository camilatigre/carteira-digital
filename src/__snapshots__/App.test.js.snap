// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deveria renderizar o App 1`] = `
<GlobalProvider>
  <BrowserRouter>
    <Switch>
      <Route
        component={[Function]}
        exact={true}
        path="/"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/register"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/home"
      />
      <Route
        component={[Function]}
        exact={true}
        path="/401"
      />
    </Switch>
  </BrowserRouter>
</GlobalProvider>
`;
